src/server.rs

```diff
@@ -209,25 +210,33 @@ async fn find_key(store: Arc<KvStore>, key: String) -> Result<impl Reply, Reject
     }
 }
 
-#[derive(Debug, Deserialize)]
-struct PatchValue {
-    operation: String,
-}
```

Comment:
Why was this removed? Adding support for other operations with more arguments than just the key using a json argument makes sense to me, even though it's just used for locking and unlocking keys right now. The structure used to do this should probably be replaced with an enum though.

Reply:
Enum seem to be a good idea

Reply:
Adding a bit more on the idea of using an enum to represent patch operations:

Different operations would just be represented as enum variants.
Serde would detect an invalid operation name or a missing parameter automatically without us having to use `Option` for everything.
IMHO the [internally tagged representation](https://serde.rs/enum-representations.html#internally-tagged) looks the best, so I suggest using that, but there's no other benefit over the externally tagged representation.

Reply:
Hmm, do we need to send JSON as the PATCH parameter?

Reply:
It doesn't have to be json specifically, but we're already using it to return information to the client and since we're using HTTP it makes more sense than something like MessagePack.

Reply:
The problem with JSON it's that a bit long for nothing, for most operations we just need one word to do something and for longer operations like TTL we surely can do that without JSON.

Honestly, I don't know what is the best solution may be a hybrid way

Reply:
I'm not sure if any good "hybrid way" exists here. Most formats that provide smaller serializations are binary and not as widespread, and I don't think we want that in our API. `{"operation":"unlock"}` is still only 22 bytes long and there's not much difference in adding or taking a few bytes. We probably aren't going to add thousands of fields to this structure, so the main bottleneck for applications is going to be network latency anyway.

An alternative is getting rid of the PATCH method entirely and adding more endpoints instead.
`PATCH /api/kv/<key>` with a json body would turn into `PUT /api/kv/<key>/lock` with a simple boolean value as a body. It would still require parsing the body, but there would be no json involved.

Reply:
Hmm. Maybe we can parse the body and if it's json we keep {"operation": "lock"}

Otherwise we perform just LOCK body request

For adding a path parameter it can be good to get objectif metas by example

Reply:
That's possible too, but as soon as we add another operation and it'll just be a really weird api quirk. There's no reason for the default action to be "lock this element" instead of any other.
It could also introduce silent errors because any request would be valid, even if the client intended to send valid json with an operation that is not `lock`.

I have no idea what you mean by "objectif metas", but we'd pretty much just be moving the operation name into the path.

Reply:
Oh my bad I mean objects meta, for the no json query I said lock as an example, but do like you feel I will adapt me and code other operations from you